### PR TITLE
Enable return address hijacking for arm64

### PR DESCRIPTION
**Pull request by rahku**

### CODE REVIEW

**Comment by rahku** [comment]
@dotnet-bot test arm64 Windows_NT checked

**Comment by rahku** [comment]
@dotnet-bot test arm64 Windows_NT checked gcstress0x3

**Comment by rahku** [comment]
PTAL @dotnet/arm64-contrib

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 Checked gcstress0x3

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test

**Comment by rahku** [comment]
@swaroop-sridhar can you PTAL
This fixes a bug in HIjackFrame::GCScanRoots due to which all GC references in multireg return were not getting promoted

**Comment by swaroop-sridhar** [comment]
Looks good, thanks for the fix

File: src/inc/gcinfotypes.h [hunk 1]
```diff
@@ -310,12 +310,13 @@ inline ReturnKind GetStructReturnKind(ReturnKind reg0, ReturnKind reg1)
...
 }
 
-inline ReturnKind ExtractRegReturnKind(ReturnKind returnKind, size_t regNo)
+inline ReturnKind ExtractRegReturnKind(ReturnKind returnKind, size_t regNo, bool& moreRegs)
 {
     _ASSERTE(IsValidReturnKind(returnKind));
     _ASSERTE(IsValidReturnRegister(regNo));
 
     ReturnKind regReturnKind = (ReturnKind)((returnKind >> (regNo * 2)) & 3);
+    moreRegs = (returnKind >> ((regNo+1) * 2)) != 0;
```
Comment (briansull) [inline src/inc/gcinfotypes.h]: A comment explaining the voodoo that is used to compute regReturnKind and set moreRegs would seem essential here.

Comment (briansull) [inline src/inc/gcinfotypes.h]: Instead of regNo, maybe call it returnRegOrdinal, and  state that the legal values are 0 or 1.

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Pull request merged**